pkg/model/filters/filters.go

```diff
@@ -52,3 +55,27 @@ func Parse(raw string) (MultiQueries, error) {
 	}
 	return parsed, nil
 }
+
+func SplitForReportersMerge(q SingleQuery) (SingleQuery, SingleQuery) {
+	// If FlowDirection is enforced, skip merging both reporters
+	for _, m := range q {
+		if m.Key == fields.FlowDirection {
+			return q, nil
+		}
+	}
+	// Merging is done by running a first query with FlowDirection=EGRESS and another with FlowDirection=INGRESS AND SrcOwnerName is empty.
+	// (Note that we use SrcOwnerName both as an optimization as it's a Loki index,
+	// and as convenience because looking for empty fields won't work if they aren't indexed)
```

Comment:
hmm never mind - topology & metrics always use raw flows, not conversations, and this code is only called from topology endpoint ... so that's fine I guess

Reply:
The metrics queries can be based on conversations when available.
As Both `FlowDirection` and `SrcOwnerName` fields are available in conversations, the approch is fine.
However:
- FlowDirection is not reinterpreted in conversations
- SrcOwnerName can actually be flipped by the swapAB option depending on TCP flags

Reply:
so .. this brings us back to swapAB needing to switch direction.. I'll open a separate bug about that

Reply:
https://issues.redhat.com/browse/NETOBSERV-1109